psydac/linalg/basic.py

```diff
@@ -899,11 +926,44 @@ def dtype(self):
 
     @property
     def linop(self):
+        """
+        The linear operator L of which this object is the inverse L^{-1}.
+
+        The linear operator L can be modified in place, or replaced entirely
+        through the setter. A substitution should only be made in cases where
+        no other options are viable, as it breaks the one-to-one map between
+        the original linear operator L (passed to the constructor) and the
+        current `InverseLinearOperator` object L^{-1}. Use with extreme care!
+
+        """
         return self._A
+    
+    @linop.setter
+    def linop(self, a):
+        assert isinstance(a, LinearOperator)
+        assert a.domain is self.domain
+        assert a.codomain is self.codomain
+        self._A = a
 
```

Comment:
I think it would also make sense to add a setter for `x0` as I don't think that so far there is an easy way to modify it and one might want to invert several time the same system with different rhs and so start with different `x0`.

EDIT : I even think we could have a setter for all parameters in `self._options` as one might want to change them for different solves

Reply:
Is this not sufficient?
```python
A_inv = inverse(A, 'cg')
A_inv.options['x0'] = x0
```

Reply:
Although I have just realized that we have a dedicated method which is safer, because it adds additional checks:
``` python
A_inv = inverse(A, 'cg')
A_inv.set_options(x0=x0)
```

Reply:
I have removed the useless (and dangerous) `options` property in 9fe92f1.

Reply:
Further, I have improved `get_options` and updated the tests in 7def827.